src/btf.h

```diff
@@ -156,13 +153,19 @@ class BTF {
   BPFtrace* bpftrace_ = nullptr;
   std::string all_funcs_;
   std::string all_rawtracepoints_;
+  bool modules_loaded_ = false;
 };
 
 inline bool BTF::has_data() const
 {
   return state == OK;
 }
 
+inline bool BTF::modules_loaded() const
+{
+  return modules_loaded_;
```

Comment:
Nit: we have the “state”, why not encode this in there? It’s always nice when the types are not capable of representing an invalid state, e.g. currently you may have uninitialized and modules_loaded_=true as an impossible state? If you encode MODULES_LOADED in the state then you model as a state machine, can assert regarding current state, etc.

Reply:
I'll look into fixing this but modules can't be loaded if the state is not already OK.